File: backend/src/modules/budget/infrastructure/model/Entry.cs
namespace backend.modules.budget.infrastructure.model;

using backend.modules.budget.domain.entry;

public class Entry
{
    public int? Id { get; set; }
    public required string Name { get; set; }
    public string? Description { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime? UpdatedAt { get; set; }
    public Category? Category { get; set; }
    public int Value { get; set; }
    public Type Type { get; set; }
}

File: frontend/src/types/Entry.ts
import type { Category } from './Category';
import { EntryType } from './EntryType';

export interface Entry {
  id: {
    value: number;
  };
  name: string;
  description: string;
  value: number;
  type: EntryType;
  category: Category | null;
  entryDate: string;
  createdAt: string;
  updatedAt: string | null;
}
